code/cfile/cfilesystem.cpp

```diff
@@ -54,14 +53,21 @@ enum CfileRootType {
 //    specifying cd-rom tree
 //    searching for pack files on CD-rom tree
 typedef struct cf_root {
-	char	path[CF_MAX_PATHNAME_LENGTH];	// Contains something like c:\projects\freespace or c:\projects\freespace\freespace.vp
+	SCP_string	path;	// Contains something like c:\projects\freespace or c:\projects\freespace\freespace.vp
 	int		roottype;						// CF_ROOTTYPE_PATH  = Path, CF_ROOTTYPE_PACK =Pack file, CF_ROOTTYPE_MEMORY=In memory
 	uint32_t location_flags;
+
+#ifdef SCP_UNIX
+	// map of existing case sensitive paths
+	SCP_unordered_map<int, SCP_string> pathTypeToRealPath;
```

Comment:
Since AIUI `SCP_UNIX` includes macOS and HFS+ is case-insensitive by default, will the `SCP_UNIX`/non-`_WIN32` changes in this PR preserve macOS compatibility?

I know FSO isn't working on macOS at the moment, but IMHO it'd be nice to avoid breaking Mac compatibility if possible.

However, if SCP wants to drop Mac support, IMO that would require a GitHub issue of its own for discussion.

Reply:
It shouldn't have any real impact on macOS. This code was already there and in-use previously, but was only used when files are initially indexed, so it wasn't applied during the other times that it might be needed. I just made it a part of the general process so that it wouldn't create a different file structure on case-sensitive filesystems (`Data` on read, `data` on write).